core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -107,36 +114,41 @@ trait WorkflowSubmission extends FutureSupport with LazyLogging with MethodWiths
 
     //flip the workflows to Launching in a separate txn.
     //if this optimistic-lock-exceptions with another txn, this one will barf and we'll reschedule when we pipe it back to ourselves
-    workflowRecsToLaunch flatMap { wfRecs =>
-      dataSource.inTransaction { dataAccess =>
-        dataAccess.workflowQuery.batchUpdateStatus(wfRecs, WorkflowStatuses.Launching)
-      } map { count =>
-        if( wfRecs.nonEmpty ) {
-          SubmitWorkflowBatch(wfRecs.map(_.id))
-        } else {
-          ScheduleNextWorkflow
+    workflowRecsToLaunch flatMap {
+      case Some((wfRecs, submissionRec, workspaceRec)) if wfRecs.nonEmpty =>
+        dataSource.inTransaction { dataAccess =>
+          workflowStatusCounter(workspaceRec.toWorkspaceName, submissionRec.id, WorkflowStatuses.Launching).countDBResult {
+            dataAccess.workflowQuery.batchUpdateStatus(wfRecs, WorkflowStatuses.Launching)
+          } map { _ =>
+            SubmitWorkflowBatch(wfRecs.map(_.id), submissionRec, workspaceRec)
+          }
         }
-      }
+      case _ => Future.successful(ScheduleNextWorkflow)
     } recover {
       // if we found some but another actor reserved the first look again immediately
-      case t: RawlsConcurrentModificationException => ProcessNextWorkflow
+      case _: RawlsConcurrentModificationException => ProcessNextWorkflow
     }
   }
 
-  def reserveWorkflowBatch(dataAccess: DataAccess, activeCount: Int)(implicit executionContext: ExecutionContext): ReadWriteAction[Seq[WorkflowRecord]] = {
+  def reserveWorkflowBatch(dataAccess: DataAccess, activeCount: Int)(implicit executionContext: ExecutionContext): ReadWriteAction[Option[(Seq[WorkflowRecord], SubmissionRecord, WorkspaceRecord)]] = {
```

Comment:
Blergh. I see why you need this - you need the submission and workspace details for `getUnlaunchedWorkflowBatch` - but this feels like a super heavyweight solution. I can't see another way around it though (sure, we could pass just the `submissionId` and `WorkspaceName` but it's still an option-of-triple).